syntax = "proto3";
import "google/protobuf/empty.proto";
option go_package=".;proto";

service User {
  rpc GetUserList(PageInfo) returns (UserListResponse);//用户列表
  rpc GetUserByMobile(MobileRequest) returns (UserInfoResponse);//通过mobile查询user
  rpc GetUserById(IdRequest) returns (UserInfoResponse);//通过id查询user
  rpc CreateUser(CreateUserInfo) returns (UserInfoResponse);//添加用户
  rpc UpdateUser(UpdateUserInfo) returns (google.protobuf.Empty);//更新用户
  rpc CheckPassword(PasswordCheck) returns (CheckPasswordResponse);//检查密码
}


message PasswordCheck {
   string password = 1;  //
  string encrypted = 2; //
}
message CheckPasswordResponse {
  bool success = 1;
}

//消息类型发送
message PageInfo {
  uint32 pageNo=1;  //第几页
  uint32 pageSize=2;//每页几个
}
message MobileRequest {
  string mobile = 1;  //电话
}
message IdRequest {
  string  id = 1;  //用户id
}
message CreateUserInfo {
  string nickName = 1;  //用户名
  string  password = 2;  //密码
  string  mobile = 3;  //手机
}
message UpdateUserInfo {
  string id = 1; //
  string nickname = 2; //
  string gender = 3; //
  uint64 birthday = 4; //
}

//响应消息类型
message UserInfoResponse {
  int32  id = 1;  //
  string password = 2;  //
  string mobile = 3; //
  string nickName = 4; //
  uint64 birthday = 5; //
  string gender = 6; //
  int32 role = 7; //
}


message UserListResponse   {
  int32 total = 1;  //共多少个用户
  repeated UserInfoResponse data =2;  //
}


